### DISCUSSION TITLE
is it possible to add cache expiration feature?

### DISCUSSION THREAD

**Discussion by ophiry** [flyteorg/flyte#3238] [Ideas]
that is - include in the task decorator a parameter which will control for how long the cache will be considered valid

**Comment by kumare3** [comment]
It is possible. It's just that where does the expiration live. If it is with the task decorator - what happens when you change it. From what I understand is - that every instance of the cache should  be associated with the cache ttl. So even if you extend or lower the ttl, it will not impact previously cached values. 

This can be confusing?

**Reply by ophiry** [reply]
I think the TTL should be controlled either on read or on write (not both - to avoid confusion)
* on write - the expiration is defined when the cached value is created
* on read - when you read the cache the TTL value is used to check if the cache didn't expire

the benefit of TTL definition on write is that it will allow clearing storage space
the benefit of TTL definition on read is that it gives more control 

I prefer the read option - but either will be OK

**Reply by kumare3** [reply]
I like the read option as well.
Cc @ophiry could you make this an issue and suggest an api design for the user facing code
